[caffe2][hipify] remove un-used flag from `pybind_utils.h`

JenniferWang opened the pull request:
Summary:
Encountered issues related to AMD build when working on https://www.internalfb.com/diff/D60739324?dst_version_fbid=2203158110057105 (see stack trace P1545717562)

Looking at the file history, seems that the flag is no longer used so I propose to remove it.  Alternatively, I could change the `#ifdef` to check both `USE_C10D_NCCL` and  `USE_ROCM` and include the corresponding AMD header files.

Let me know what is more preferred way.

Test Plan: Sandcastle

Differential Revision: D61762129

JenniferWang commented:
/easycla

malfet reviewed (approved):
If it compiles, than sounds good to me

JenniferWang committed:
[caffe2][hipify] remove un-used flag from `pybind_utils.h` (#134404)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/134404

Encountered issues related to AMD build when working on https://www.internalfb.com/diff/D60739324?dst_version_fbid=2203158110057105 (see stack trace P1545717562)

Looking at the file history, seems that the flag is no longer used so I propose to remove it.  Alternatively, I could change the `#ifdef` to check both `USE_C10D_NCCL` and  `USE_ROCM` and include the corresponding AMD header files.

Let me know what is more preferred way.

Test Plan: Sandcastle

Reviewed By: malfet

Differential Revision: D61762129
